features/contract-state/GeneralInfo.tsx

```diff
@@ -44,101 +44,127 @@ const GeneralInfo = () => {
   } = empState;
   const { symbol: tokenSymbol } = Token.useContainer();
 
-  // format nice date
-  const expiryDate = expiry ? new Date(expiry.toNumber() * 1000) : "N/A";
-
-  const pricedGcr = gcr && latestPrice ? gcr / Number(latestPrice) : null;
-
-  const withdrawalLivenessInMinutes = withdrawalLiveness
-    ? Number(withdrawalLiveness) / 60
-    : null;
-
-  return (
-    <Box>
-      <Typography variant="h5">
-        General Info{" "}
-        {contract?.address && (
-          <Link
-            href={getEtherscanUrl(contract.address)}
-            target="_blank"
-            rel="noopener noreferrer"
+  if (
```

Comment:
@mrice32 @chrismaree @adrianmcli 

I think I also came up with a way to display "stateless" components and maximize "discoverability. 

If you scroll down to the end of this file you'll see a function `renderComponent` that takes in data to display. This is the stateless component and should work even if the data is "missing" because we define default values for it.

Back to this `if` statement logic: We will set a series of data variables to display IFF all of the data is valid and in the expected form. If all of the data is ready then we'll pass on the overidden values to `renderComponent()` otherwise we'll just call `renderComponent()` with no arguments and let the defaults carry over.